### ISSUE TITLE
Bubble Heads

### ISSUE THREAD

**Issue by amark** [meething/webrtc-gun#26]
@QVDev disagrees with this 👍 

@lmangani agrees but think not possible in 8 weeks 👍 .

Mark is excited by this, haha 👎 

Check out: 

 - https://www.intercom.com/ 
 - https://www.userlike.com/en/features

Goal:

 - Drop in 1 script tag on a website, optionally set some configs "autoconnect: false" etc.
 - Bottom right bubble shows current person who is talking, so it switches between people as talking.
 - Hover over the bubble and it spreads all the bubbles out across the screen so you can see all talking people.
 - Click on any person and they become full screen background ontop of the website but below the other bubbles
 - Drag any bubble around the screen to re-arrange.

Can answer any other Qs or fill in more info.

**Comment by amark** [comment]
Revision for @QVDev 👍 

 - Config of `autofullscreen: true`.

This would be used on meething website itself, to show default UI fully expanded.

**Comment by Dletta** [comment]
This could be done using d3, I believe, basically use the underlying force layout used in, let's say my graph visualizer, and just have the heads as disconnected nodes (they are svg / html elements)

**Comment by amark** [comment]
D3 is cool, I love it. tho probably overkill here:

For example, this can be done in 30 lines (no comments) with pretty much 99% CSS (just tad JS to update CSS on drag)

https://codepen.io/chengwei9993/pen/PoqNxqR

**Comment by lmangani** [comment]
I do not think this can easily fall into [our scope](https://en.wikipedia.org/wiki/KISS_principle) of connecting people in a video room, and the best we can do in current usecase as discussed would be creating ONE div with cool styling and one or many participants bubbles to pop out using the PIP functionality of the browsers.

This said, the very same tech we are stabilizing could be recycled very easily to create features like intercom or whatnot as a separate app with this specific purpose in mind. Sounds like something Iris perhaps should incorporate?

**Comment by amark** [comment]
PIP sounds cool, I haven't heard of this browser feature/ability yet/before! Googling... mostly see extensions for this? Have a better link (no need tho)?

**Comment by lmangani** [comment]
Our demo already implements the basic form... just click on any of the video streams or your local video. We're hoping this can be styled and of course, we can PIP a frame with many people too.

**Comment by lmangani** [comment]
How to fake a video frame for PIP using canvas content: https://blog.agney.dev/pomodoro-on-pip/

**Comment by Dletta** [comment]
Here is the code for PIP that can go outside of browser https://googlechrome.github.io/samples/picture-in-picture/

**Comment by lmangani** [comment]
@Dletta pip is already implemented on all streams ;) doubleclick any of them to try

**Comment by Dletta** [comment]
Nice, you guys work so fast! Hard to keep up!

**Comment by kcsf** [comment]
love this idea.  in particular for front-end ui for meething, "bubble heads" should allow for more active participants w/multicast, while making for a more visually appealing interface for end users.  user controls can also be more effectively incorporated in hover elements around user bubbles.  

The best example we've seen in play for this is https://www.glowbl.com/
(built on Neo4j & Janus)
